Content.Server/AirlockPainter/AirlockPainterSystem.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            if (!EntityManager.TryGetComponent(args.User, out ActorComponent? actor))
+                return;
+            component.Owner.GetUIOrNull(AirlockPainterUiKey.Key)?.Open(actor.PlayerSession);
+            args.Handled = true;
+        }
+
+        private void AfterInteractOn(EntityUid uid, AirlockPainterComponent component, AfterInteractEvent args)
+        {
+            if (component.IsSpraying || args.Target is not { Valid: true } target || !args.CanReach)
+                return;
+
+            if (!EntityManager.TryGetComponent<PaintableAirlockComponent>(target, out var airlock))
+                return;
+
+            string? sprite = GetAirlockSpritePath(airlock.Group, component.Style);
+            if (sprite == null)
+            {
+                string msg = Loc.GetString("airlock-painter-style-not-available");
+                _popupSystem.PopupEntity(msg, args.User, Filter.Entities(args.User));
+                return;
+            }
+            component.IsSpraying = true;
+            var doAfterEventArgs = new DoAfterEventArgs(args.User, component.SprayTime, default, target)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true,
+                BroadcastFinishedEvent = new AirlockPainterDoAfterComplete(uid, target, sprite, component),
+                BroadcastCancelledEvent = new AirlockPainterDoAfterCancelled(component),
+            };
+            _doAfterSystem.DoAfter(doAfterEventArgs);
+        }
+
+        private sealed class AirlockPainterDoAfterComplete : EntityEventArgs
+        {
+            public readonly EntityUid User;
+            public readonly EntityUid Target;
+            public readonly string Sprite;
+            public readonly AirlockPainterComponent Component;
+
+            public AirlockPainterDoAfterComplete(EntityUid user, EntityUid target, string sprite, AirlockPainterComponent component)
+            {
+                User = user;
+                Target = target;
+                Sprite = sprite;
+                Component = component;
+            }
+        }
+
+        private sealed class AirlockPainterDoAfterCancelled : EntityEventArgs
+        {
+            public readonly AirlockPainterComponent Component;
+
+            public AirlockPainterDoAfterCancelled(AirlockPainterComponent component)
+            {
+                Component = component;
+            }
+        }
+
+        private void OnSpritePicked(EntityUid uid, AirlockPainterComponent component, AirlockPainterSpritePickedMessage args)
+        {
+            component.Style = (AirlockStyle)args.Index;
+        }
+
+        private static string? GetAirlockSpritePath(AirlockGroup grp, AirlockStyle style)
+        {
+            // This should probably be put into a data file
```

Comment:
this should definitely be in yaml, yeah. maybe an `AirlockGroupPrototype` that has a `Dictionary<string, string>` for the style -> RSI path data, where the string key is the ID for an `AirlockStylePrototype` or smth. then instead of doing `Enum.GetValues<T>` you can do `IPrototypeManager.EnumeratePrototypes<T>`